app/assets/javascripts/admin/addon/services/admin-search-data-source.js

```diff
@@ -0,0 +1,317 @@
+        if (
+          mapItem.keywords.match(escapedFilterRegExp) &&
+          typeItemCount <= perTypeLimit
+        ) {
+          filteredResults.push(mapItem);
+          typeItemCount++;
+        }
+      });
+    });
+
+    return filteredResults;
+  }
+
+  #addPageLink(mapItem, link, parentLabel = "") {
+    let url;
+    if (link.routeModels) {
+      url = this.router.urlFor(link.route, ...link.routeModels);
+    } else {
+      url = this.router.urlFor(link.route);
+    }
+
+    const mapItemLabel = this.#labelOrText(mapItem);
+    const linkLabel = this.#labelOrText(link);
+
+    let label;
+    if (parentLabel) {
+      label = mapItemLabel;
+      if (mapItemLabel) {
+        label += ` ${SEPARATOR} `;
+      }
+      label += `${parentLabel} ${SEPARATOR} ${linkLabel}`;
+    } else {
+      label = mapItemLabel + (mapItemLabel ? ` ${SEPARATOR} ` : "") + linkLabel;
+    }
+
+    if (link.settings_area) {
+      this.settingPageMap.areas[link.settings_area] = link.multi_tabbed
+        ? `${url}/settings`
+        : url;
+    }
+
+    if (link.settings_category) {
+      this.settingPageMap.categories[link.settings_category] = link.multi_tabbed
+        ? `${url}/settings`
+        : url;
+    }
+
+    const linkKeywords = link.keywords
+      ? i18n(link.keywords).toLowerCase().replaceAll("|", " ")
+      : "";
+    const linkDescription = link.description
+      ? link.description.includes(" ")
+        ? link.description
+        : i18n(link.description)
+      : "";
+
+    this.pageMapItems.push({
+      label,
+      url,
+      keywords: this.#buildKeywords(
+        linkKeywords,
+        url,
+        label.replace(SEPARATOR, "").toLowerCase().replace(/  +/g, " "),
+        linkDescription
+      ),
+      type: "page",
+      icon: link.icon,
+      description: linkDescription,
+    });
+
+    return linkLabel;
+  }
+
+  #processSettings(settings) {
+    const settingPluginNames = {};
+
+    settings.forEach((setting) => {
+      let plugin;
+
+      let rootLabel;
+      if (setting.plugin) {
+        if (!settingPluginNames[setting.plugin]) {
+          settingPluginNames[setting.plugin] = setting.plugin.replaceAll(
+            "_",
+            "-"
+          );
+        }
+
+        plugin = this.plugins[settingPluginNames[setting.plugin]];
+
+        if (plugin) {
+          rootLabel = plugin.admin_route?.label
+            ? i18n(plugin.admin_route?.label)
+            : i18n("admin.plugins.title");
+        } else {
+          rootLabel = i18n("admin.plugins.title");
+        }
+      } else if (setting.primary_area) {
+        rootLabel =
+          I18n.lookup(`admin.config.${setting.primary_area}.title`) ||
+          i18n(`admin.site_settings.categories.${setting.category}`);
+      } else {
+        rootLabel = i18n(`admin.site_settings.categories.${setting.category}`);
+      }
+
+      const label = `${rootLabel} ${SEPARATOR} ${setting.setting}`;
```

Comment:
Can we use the nicely formatted setting name like we do in the settings page?

<img width="276" alt="Screenshot 2025-02-20 at 11 25 53 AM" src="https://github.com/user-attachments/assets/e4a17dfa-c7f2-4fcf-8c04-f7ef57cd4c93" />

Reply:
Oh yes good point!